### PR TITLE
Move raise_fn_clause_error check to common checks

### LINKED ISSUE #156: Common check for `raise FunctionClauseError`
Some of the unit tests for learning exercises (and collatz-conjecture) contain assertion that specific function calls raise `FunctionClauseError`. This is to check that the student didn't implement too permissive function clauses.

However, some students take that too literally and actually write `def foo(_), do: raise FunctionClauseError`.

There should be a common check for every solution that `raise FunctionClauseError` is never called. AFAIK there isn't ever a need to raise this error explicitly (at least on Exercism). 

It should be possible to implement this by using the `feature / form` macros, similar to how `ElixirAnalyzer.ExerciseTest.CommonChecks.DebugFunctions` was implemented with the `assert_no_call` macro.

### PR DESCRIPTION
**Pull request by Br1ght0ne**
Close #156.

### CODE REVIEW

File: lib/elixir_analyzer/exercise_test/common_checks/uncommon_errors.ex [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  defmacro __using__(_opts) do
+    quote do
+      feature "raises function clause error" do
+        find :none
+        type :actionable
+        comment ElixirAnalyzer.Constants.solution_raise_fn_clause_error()
+
+        form do
+          raise FunctionClauseError
+        end
```
Comment (Br1ght0ne) [inline lib/elixir_analyzer/exercise_test/common_checks/uncommon_errors.ex:15]: Any other forms that need to be handled?

Comment (angelikatyborska) [reply lib/elixir_analyzer/exercise_test/common_checks/uncommon_errors.ex:15]: One more. `raise` can also be used with structs:
```elixir
iex(6)> raise %FunctionClauseError{}
** (FunctionClauseError) no function clause matches
```

File: test/elixir_analyzer/test_suite/two_fer_test.exs [hunk 2]
```diff
@@ -209,19 +209,4 @@ defmodule ElixirAnalyzer.TestSuite.TwoFerTest do
...
-        def two_fer(name \\ "you") do
-          if is_bitstring(name) do
-            "One for #{name}, one for me."
-          else
-            raise FunctionClauseError
-          end
-        end
-      end
-    end
-  end
```
Comment (angelikatyborska) [inline test/elixir_analyzer/test_suite/two_fer_test.exs:226]: It's a good choice to remove the exercise-specific test, but we need to have some sort of unit test for this check. Could you add something like in this PR https://github.com/exercism/elixir-analyzer/pull/153/files (`test/elixir_analyzer/exercise_test/common_checks/last_line_assignment_test.exs`)?

**Review by angelikatyborska — COMMENTED** [review]
Nice! I wasn't aware that this check already exists for Two-Fer. The existing comment for this check is general so it will still work after moving the check to common checks 🙂

**Pull request merged**